// @morev/utils/dist/chunk-M5HBLC4T.js
// src/dates/format-date/format-date.locales.ts
var FORMAT_DATE_LOCALE_EN = {
  name: "en",
  values: {
    firstDayOfWeekIndex: 0,
    dayNames: {
      abbreviated: ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
      wide: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
    },
    monthNames: {
      abbreviated: ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"],
      wide: ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
    }
  }
};
var FORMAT_DATE_LOCALE_RU = {
  name: "ru",
  values: {
    firstDayOfWeekIndex: 1,
    dayNames: {
      abbreviated: ["\u0412\u0441", "\u041F\u043D", "\u0412\u0442", "\u0421\u0440", "\u0427\u0442", "\u041F\u0442", "\u0421\u0431"],
      wide: [
        ["\u0412\u043E\u0441\u043A\u0440\u0435\u0441\u0435\u043D\u044C\u0435", "\u0412 \u0432\u043E\u0441\u043A\u0440\u0435\u0441\u0435\u043D\u044C\u0435"],
        ["\u041F\u043E\u043D\u0435\u0434\u0435\u043B\u044C\u043D\u0438\u043A", "\u0412 \u043F\u043E\u043D\u0435\u0434\u0435\u043B\u044C\u043D\u0438\u043A"],
        ["\u0412\u0442\u043E\u0440\u043D\u0438\u043A", "\u0412\u043E \u0432\u0442\u043E\u0440\u043D\u0438\u043A"],
        ["\u0421\u0440\u0435\u0434\u0430", "\u0412 \u0441\u0440\u0435\u0434\u0443"],
        ["\u0427\u0435\u0442\u0432\u0435\u0440\u0433", "\u0412 \u0447\u0435\u0442\u0432\u0435\u0440\u0433"],
        ["\u041F\u044F\u0442\u043D\u0438\u0446\u0430", "\u0412 \u043F\u044F\u0442\u043D\u0438\u0446\u0443"],
        ["\u0421\u0443\u0431\u0431\u043E\u0442\u0430", "\u0412 \u0441\u0443\u0431\u0431\u043E\u0442\u0443"]
      ]
    },
    monthNames: {
      abbreviated: ["\u042F\u043D\u0432", "\u0424\u0435\u0432", "\u041C\u0430\u0440\u0442", "\u0410\u043F\u0440", "\u041C\u0430\u0439", "\u0418\u044E\u043D\u044C", "\u0418\u044E\u043B\u044C", "\u0410\u0432\u0433", "\u0421\u0435\u043D", "\u041E\u043A\u0442", "\u041D\u043E\u044F", "\u0414\u0435\u043A"],
      wide: [
        ["\u042F\u043D\u0432\u0430\u0440\u044C", "\u042F\u043D\u0432\u0430\u0440\u044F"],
        ["\u0424\u0435\u0432\u0440\u0430\u043B\u044C", "\u0424\u0435\u0432\u0440\u0430\u043B\u044F"],
        ["\u041C\u0430\u0440\u0442", "\u041C\u0430\u0440\u0442\u0430"],
        ["\u0410\u043F\u0440\u0435\u043B\u044C", "\u0410\u043F\u0440\u0435\u043B\u044F"],
        ["\u041C\u0430\u0439", "\u041C\u0430\u044F"],
        ["\u0418\u044E\u043D\u044C", "\u0418\u044E\u043D\u044F"],
        ["\u0418\u044E\u043B\u044C", "\u0418\u044E\u043B\u044F"],
        ["\u0410\u0432\u0433\u0443\u0441\u0442", "\u0410\u0432\u0433\u0443\u0441\u0442\u0430"],
        ["\u0421\u0435\u043D\u0442\u044F\u0431\u0440\u044C", "\u0421\u0435\u043D\u0442\u044F\u0431\u0440\u044F"],
        ["\u041E\u043A\u0442\u044F\u0431\u0440\u044C", "\u041E\u043A\u0442\u044F\u0431\u0440\u044F"],
        ["\u041D\u043E\u044F\u0431\u0440\u044C", "\u041D\u043E\u044F\u0431\u0440\u044F"],
        ["\u0414\u0435\u043A\u0430\u0431\u0440\u044C", "\u0414\u0435\u043A\u0430\u0431\u0440\u044F"]
      ]
    }
  }
};

export {
  FORMAT_DATE_LOCALE_EN,
  FORMAT_DATE_LOCALE_RU
};
